sklbench/utils/env.py

```diff
@@ -0,0 +1,145 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import json
+from typing import Dict
+
+import pandas as pd
+
+from .common import read_output_from_command
+from .logger import logger
+
+
+def get_numa_cpus_conf() -> Dict[int, str]:
+    try:
+        _, lscpu_text, _ = read_output_from_command("lscpu")
+        return {
+            i: numa_cpus
+            for i, numa_cpus in enumerate(
+                map(
+                    lambda x: x.split(" ")[-1],
+                    filter(
+                        lambda line: "NUMA" in line and "CPU(s)" in line,
+                        lscpu_text.split("\n"),
+                    ),
+                )
+            )
+        }
+    except FileNotFoundError:
+        logger.warning("Unable to get numa cpus configuration via lscpu")
+        return dict()
+
+
+def get_software_info() -> Dict:
+    result = dict()
+    # conda list
+    try:
+        _, conda_list, _ = read_output_from_command("conda list --json")
+        conda_packages = json.loads('{"list":\n' + conda_list + "\n}")["list"]
+        result["conda_packages"] = {pkg.pop("name"): pkg for pkg in conda_packages}
+    except (FileNotFoundError, PermissionError):
```

Comment:
```suggestion
    except (FileNotFoundError, PermissionError, AttributeError):
```
AttributeError is also possible there if output of conda list contains error message because in this case output json is depth 1 and its values have no pop attribute.